### PR TITLE
gh-124058: remove _PyCompile_IsNestedScope, roll it into _PyCompile_IsInteractive

### LINKED ISSUE #124058: Remove `_PyCompile_IsNestedScope` roll it into `_PyCompile_IsInteractive`
Currently the `c_interactive` field is used in only one place - to special-case the code generation of interactive statements. We can move this special case from `codegen_stmt_expr` to `_PyCodegen_Body` (where there is the `is_interactive` arg) and then we no longer need `c_interactive`, its accessor function _PyCompile_IsInteractive, and the `_PyCompile_IsNestedScope` function.

### Linked PRs
* gh-124061

### PR DESCRIPTION
**Pull request by iritkatriel**
Fixes #124058.

* Issue: gh-124058

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
```
jelle@m3mb-jelle cpython % ./python.exe         
Python 3.14.0a0 (heads/move_interactive:cebdf71c554, Sep 13 2024, 09:10:41) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 1; 2Assertion failed: (asdl_seq_LEN(stmts) == 1), function codegen_interactive, file codegen.c, line 748.
                                                                                                             zsh: abort      ./python.exe
```

This doesn't work correctly if there are multiple statements in one interactive.

**Comment by iritkatriel** [comment]
I don't get it:

```
>>> dis.dis(compile("1;2", "", "single"))
  0           RESUME                   0

  1           LOAD_CONST               0 (1)
              CALL_INTRINSIC_1         1 (INTRINSIC_PRINT)
              POP_TOP
              LOAD_CONST               1 (2)
              CALL_INTRINSIC_1         1 (INTRINSIC_PRINT)
              POP_TOP
              RETURN_CONST             2 (None)
>>> 1;2
2
```

how come the 1 is not printed?

**Comment by iritkatriel** [comment]
> I don't get it:
> 
> ```
> >>> dis.dis(compile("1;2", "", "single"))
>   0           RESUME                   0
> 
>   1           LOAD_CONST               0 (1)
>               CALL_INTRINSIC_1         1 (INTRINSIC_PRINT)
>               POP_TOP
>               LOAD_CONST               1 (2)
>               CALL_INTRINSIC_1         1 (INTRINSIC_PRINT)
>               POP_TOP
>               RETURN_CONST             2 (None)
> >>> 1;2
> 2
> ```
> 
> how come the 1 is not printed?

This does print the 1:

```
>>> exec(compile("1;2", "", "single"))
1
2
```

so at the prompt we're doing something else.

File: Python/compile.c [hunk 1]
```diff
@@ -1205,16 +1205,11 @@ _PyCompile_OptimizationLevel(compiler *c)
 
 int
 _PyCompile_IsInteractive(compiler *c)
```
Comment (JelleZijlstra) [inline Python/compile.c]: What do you think about renaming this to something like `_PyCompile_IsInteractiveTopLevel` for clarity?

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**